File: Lessons/4-counting_elements/perm_check/solution.go
package solution

// you can also use imports, for example:
// import "fmt"
// import "os"

// you can write to stdout for debugging purposes, e.g.
// fmt.Println("this is a debug message")

func Solution(A []int) int {
    // write your code in Go 1.4
    n := len(A)
    counters := make([]bool, n)

    notPerm := 0
    perm := 1

    for _, value := range A {
        if value > n {
            return notPerm
        }
        if counters[value - 1] {
            return notPerm
        }

        counters[value - 1] = true
    }

    return perm
}

File: lessons/4-counting_elements/perm_check/solution.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(a)
  # write your code in Ruby 2.2
  n = a.length
  counters = Array.new n, false
  is_not_permutation = 0
  is_permutation = 1

  a.each do |value|
    return is_not_permutation if value > n
    return is_not_permutation if counters[value - 1]

    counters[value - 1] = true
  end

  is_permutation
end
